content/general/concepts/algorithm/terms/euclidean-algorithm/kadane-algorithm.md

```diff
@@ -0,0 +1,81 @@
+---
+Title: 'Kadane's Algorithm'
+Description: 'Returns the maximum sum of a subarray within an array of numbers'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Algorithms'
+  - 'Arithmetic'
+  - 'Arrays'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+Kadane's algorithm is an efficient way to find the maximum sum of a contiguous subarray within a given array of numbers. It iterates through the array and keeps track of the current maximum sum and the maximum sum found so far. By comparing the current element with the running sum, it decides whether to continue the subarray or start a new one. The algorithm guarantees a time complexity of O(n), making it an optimal solution for this problem.
```

Comment:
This actually isn't a very clear explanation of Kadane's algorithm. Can you reword it?